### PR TITLE
[sensorfw] get compass working Fixes MER#882

### PR DESCRIPTION
**Pull request by lpotter**
* change magcalibration chain expected data type to
CalibratedMagneticFieldData

* add MagCoordinateAlignFilter to remap mag data points when needed

* invalidate the hybris adaptor if no sensor is dynamically found.

* if orientationadaptor (calibrated compass) is valid use it, otherwise
use sensorfw for calibration and compass routines.

This helps for hybris based systems, when virtual sensors are not
present. It keeps the default hybris configuration file usable.

### CODE REVIEW

**Comment by saukko** [comment]
Need a bug reference at least.

File: adaptors/hybrisaccelerometer/hybrisaccelerometeradaptor.cpp [hunk 1]
```diff
@@ -55,13 +55,13 @@ void HybrisAccelerometerAdaptor::stopSensor()
...
-
     AccelerationData *d = buffer->nextSlot();
     d->timestamp_ = quint64(data.timestamp * .001);
     // sensorfw wants milli-G'
-    d->x_ = -(data.data[0] / 9.80665 * 1000);
-    d->y_ = -(data.data[1] / 9.80665 * 1000);
-    d->z_ = -(data.data[2] / 9.80665 * 1000);
+    d->x_ = data.acceleration.x / 9.80665 * 1000;
+    d->y_ = data.acceleration.y / 9.80665 * 1000;
+    d->z_ = data.acceleration.z / 9.80665 * 1000;
```
Comment (saukko) [inline adaptors/hybrisaccelerometer/hybrisaccelerometeradaptor.cpp]: Maybe add a variable for this 9.80665 value, GRAV_ACCEL or something?

File: adaptors/hybrisgyroscopeadaptor/hybrisgyroscopeadaptor.cpp [hunk 2]
```diff
@@ -61,10 +63,10 @@ void HybrisGyroscopeAdaptor::processSample(const sensors_event_t& data)
...
     TimedXyzData *d = buffer->nextSlot();
     d->timestamp_ = quint64(data.timestamp * .001);
-    d->x_ = (data.acceleration.x) * 57295.7795;
-    d->y_ = (data.acceleration.y) * 57295.7795;
-    d->z_ = (data.acceleration.z) * 57295.7795;
-
+    d->x_ = (data.gyro.x) * RADIANS_TO_DEGREES * 1000;
+    d->y_ = (data.gyro.y) * RADIANS_TO_DEGREES * 1000;
+    d->z_ = (data.gyro.z) * RADIANS_TO_DEGREES * 1000;
+qDebug() << Q_FUNC_INFO << d->x_ << d->y_ << d->z_ << data.gyro.status;
```
Comment (saukko) [inline adaptors/hybrisgyroscopeadaptor/hybrisgyroscopeadaptor.cpp]: Indent it not correct.

File: chains/compasschain/compassfilter.cpp [hunk 3]
```diff
@@ -7,7 +7,7 @@
 
    Sensord is free software; you can redistribute it and/or modify
    it under the terms of the GNU Lesser General Public License
-   version 2.1 as published by the Free Software Foundation.
+   version 2.1 as published magY the Free Software Foundation.
```
Comment (saukko) [inline chains/compasschain/compassfilter.cpp]: search and replace? ;)

File: chains/compasschain/compassfilter.cpp [hunk 4]
```diff
@@ -38,25 +40,33 @@ CompassFilter::CompassFilter() :
...
+    qreal normalizedY = y / divisor;
+    qreal normalizedZ = z / divisor;
+//    qDebug() << Q_FUNC_INFO <<"normalized"<< normalizedX << normalizedY << normalizedZ;
+
+
     ///////////////
-    /// \brief this algorithm is from Circuit Cellar Aug 2012
-    ///  by Mark Pedley
+    /// this algorithm is from Circuit Cellar Aug 2012
+    ///  magY Mark Pedley
```
Comment (saukko) [inline chains/compasschain/compassfilter.cpp]: search and replace again ;)

File: chains/compasschain/compassfilter.cpp [hunk 5]
```diff
@@ -64,58 +74,40 @@ void CompassFilter::accelDataAvailable(unsigned, const AccelerationData *data)
...
+    normalizedZ = normalizedY * sinAngle + normalizedZ * cosAngle;
 
     /* calculate pitch angle Theta (-90deg, 90deg) and sin, cos*/
-    The = atan(-Gx / Gz) * RADIANS_TO_DEGREES;  /* Equation 3 */
-    sinAngle = qSin(The * DEGREES_TO_RADIANS); /* sin(Theta) */
-    cosAngle = qCos(The * DEGREES_TO_RADIANS); /* cos(Theta) */
+    The = atan(-normalizedX / normalizedZ);  /* Equation 3 */
+    sinAngle = qSin(The);
+    cosAngle = qCos(The);
+  //  qDebug() << Q_FUNC_INFO << "roll" << Phi<< "pitch" << The;
```
Comment (saukko) [inline chains/compasschain/compassfilter.cpp]: Indent

File: filters/magcoordinatealignfilter/magcoordinatealignfilter.cpp [hunk 6]
```diff
@@ -0,0 +1,46 @@
+/**
+   @file magMagCoordinateAlignFilter.cpp
+   @brief MagMagCoordinateAlignFilter
+
+   <p>
+   Copyright (C) 2009-2010 Nokia Corporation
+
+   @author Timo Rongas <ext-timo.2.rongas@nokia.com>
+   @author Antti Virtanen <antti.i.virtanen@nokia.com>
```
Comment (saukko) [inline filters/magcoordinatealignfilter/magcoordinatealignfilter.cpp]: Correct authors?

File: sensors/magnetometersensor/magnetometerscalefilter.cpp [hunk 7]
```diff
@@ -30,7 +30,7 @@
 MagnetometerScaleFilter::MagnetometerScaleFilter() :
         Filter<CalibratedMagneticFieldData, MagnetometerScaleFilter, CalibratedMagneticFieldData>(this, &MagnetometerScaleFilter::filter)
 {
-    factor = Config::configuration()->value("magnetometer/scale_coefficient", QVariant(300)).toInt();;
+    factor = Config::configuration()->value("magnetometer/scale_coefficient", QVariant(1)).toInt();
```
Comment (saukko) [inline sensors/magnetometersensor/magnetometerscalefilter.cpp]: Magic number 1?

File: filters/magcoordinatealignfilter/magcoordinatealignfilter.cpp (hunk 6 above)
Comment (lpotter) [inline filters/magcoordinatealignfilter/magcoordinatealignfilter.cpp]: Yes. This file is basically a copy of CoordinateAlignFilter with a datatype that supports magnetometer raw values and calibration level, so they can be passed on through the chain.

File: sensors/magnetometersensor/magnetometerscalefilter.cpp (hunk 7 above)
Comment (lpotter) [inline sensors/magnetometersensor/magnetometerscalefilter.cpp]: hybris mag sensor values are proper values that incorporate scaling factor by the time it reaches here.

File: filters/magcoordinatealignfilter/magcoordinatealignfilter.cpp [hunk 8]
```diff
@@ -0,0 +1,46 @@
+/**
+   @file magMagCoordinateAlignFilter.cpp
+   @brief MagMagCoordinateAlignFilter
```
Comment (martinjones) [inline filters/magcoordinatealignfilter/magcoordinatealignfilter.cpp]: Typo in the 2 lines above.

**Pull request merged**